changelogs/unreleased/4723-fix--override-pip-extra-index-url.yml

```diff
@@ -0,0 +1,6 @@
+---
+description: make sure that the index present in PIP_EXTRA_INDEX_URL is not leaked to the pip install_from_index
```

Comment:
I think this bug exists in the stable release. Shouldn't we add a bugfix changelog entry?